code/datums/controllers/action_controls.dm

```diff
@@ -661,12 +661,22 @@ var/datum/action_controller/actions
 		. = ..()
 		if(QDELETED(sheet) || sheet.amount < cost)
 			interrupt(INTERRUPT_ALWAYS)
+		if (ismob(owner))
+			var/mob/M = owner
+			if(!equipped_or_holding(sheet, M))
+				interrupt(INTERRUPT_ALWAYS)
```

Comment:
Couldn't you just do `equipped_or_holding(sheet, owner)` directly?

Reply:
was getting errors with that approach as `owner` is typed as an atom, so casting here is needed as `equipped` is a proc on `mob`

Reply:
that's not how casting works, the ismob check is sufficient, the cast does nothing

Reply:
when i don't cast it to a mob i get an error on build:
```code\datums\controllers\action_controls.dm:665:error: owner.equipped: undefined proc```

Reply:
did this in [eaa19ba](https://github.com/goonstation/goonstation/pull/10692/commits/eaa19bac6d5d80731f0d36a6826da3810c9382c9)

Reply:
beepsky mad :( https://github.com/goonstation/goonstation/actions/runs/3074569633/jobs/4967432880
```code/datums/controllers/action_controls.dm, line 665, column 8:
error: undefined proc: "equipped" on /atom```

where am I going wrong here?

Reply:
the cast to mob is required because we don't use `:` everywhere